tests/Integration/IntegrationTest.php

```diff
@@ -0,0 +1,178 @@
+    {
+        setlocale(LC_ALL, 'en_US.utf8');
+    }
+
+    /**
+     * @param String[] $compareFiles
+     * @dataProvider getIntegrationTests
+     */
+    public function testIntegration(
+        string $inputPath,
+        string $expectedPath,
+        string $outputPath,
+        array $compareFiles
+    ): void {
+        system("rm -rf " . escapeshellarg($outputPath));
+        system("mkdir " . escapeshellarg($outputPath));
+
+        $metas = new Metas([]);
+        $logger = new class($outputPath) extends AbstractLogger {
+
+            private $outputPath;
+
+            public function __construct($outputPath) {
+                $this->outputPath = $outputPath;
+            }
+
+            public function log($level, $message, array $context = []): void
+            {
+                $message = $level . ':' . $message . PHP_EOL;
+                file_put_contents($this->outputPath . '/log' . $level . '.txt', $message, FILE_APPEND);
+            }
+        };
+
+        $commandbus = QuickStart::create(
+            [
+                ParseFileCommand::class => new ParseFileHandler(
+                    $logger,
+                    new class implements EventDispatcherInterface
+                    {
+                        public function dispatch(object $event)
+                        {
+                            return $event;
+                        }
+                    },
+                    \phpDocumentor\Guides\Setup\QuickStart::createRstParser()
+                )
+            ]
+        );
+
+        $parseDirectoryHandler = new ParseDirectoryHandler(
+            new FileCollector($metas),
+            $commandbus,
+        );
+        $sourceFileSystem = new Filesystem(new Local(
+            $inputPath
+        ));
+        $sourceFileSystem->addPlugin(new Finder());
+
+        $parseDirCommand = new ParseDirectoryCommand(
+            $sourceFileSystem,
+            '',
+            'rst'
+        );
+
+        $documents = $parseDirectoryHandler->handle($parseDirCommand);
+        $compliler = new Compiler([
+            new MetasPass($metas),
+            new TransformerPass(
+                new DocumentNodeTraverser(
+                    [
+                        new TocNodeTransformer($metas)
+                    ]
+                )
+            )
+        ]);
+        $documents = $compliler->run($documents);
+
+        $renderer = \phpDocumentor\Guides\Setup\QuickStart::createRenderer($metas);
+        $renderDocumentHandler = new RenderDocumentHandler($renderer);
+
+        foreach ($documents as $document) {
+            try {
+                $renderDocumentHandler->handle(
+                    new RenderDocumentCommand(
+                        $document,
+                        RenderContext::forDocument(
+                            $document,
+                            $sourceFileSystem,
+                            new Filesystem(new Local($outputPath)),
```

Comment:
If we would replace this local adapter with a memory adapter, we could get rid of the removal statements at the start of this test. 

https://flysystem.thephpleague.com/v1/docs/adapter/memory/

It will require some adjustments to the asserts, as the files are not on disk anymore, but this will prevent issues when the tests are failing for some reason.

Reply:
I would like the integration tests to be as close to the real thing as possible and this also means writing files to disk.

Once you moved forward with the standalone application I would like to use the same logic here. I would rather have a cleanup make method deleting all the temp folders.